Exception thrown for using recycled HTTP request

**Steps to reproduce**

No clear steps, it is related to an old code that tries to get the remote user from the HTTP request, this worked for old Tomcat versions. But since the upgrade to Tomcat 10 and since there is no Security manager configured, the usage of the request object is not permitted after the timeout.
The following exception is thrown each time a code tries to get information from the request :
```
Caused by: java.lang.IllegalStateException: The request object has been recycled and is no longer associated with this facade
	at org.apache.catalina.connector.RequestFacade.checkFacade(RequestFacade.java:855)
	at org.apache.catalina.connector.RequestFacade.getRemoteUser(RequestFacade.java:585)
	at jakarta.servlet.http.HttpServletRequestWrapper.getRemoteUser(HttpServletRequestWrapper.java:165)
	at org.exoplatform.portal.application.PortalRequestContext.getRemoteUser(PortalRequestContext.java:567)
	at org.exoplatform.portal.application.PortalRequestContext.getUserPortalConfig(PortalRequestContext.java:299)
	at org.exoplatform.portal.application.PortalRequestContext.getPortalOwner(PortalRequestContext.java:523)
	at org.exoplatform.commons.utils.CommonsUtils.getCurrentPortalOwner(CommonsUtils.java:204)

```

--- a/webui/portal/pom.xml
+++ b/webui/portal/pom.xml
@@ -69,5 +69,9 @@
       <groupId>${project.groupId}</groupId>
       <artifactId>exo.portal.component.api</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.exoplatform.core</groupId>
+      <artifactId>exo.core.component.security.core</artifactId>
+    </dependency>
   </dependencies>
 </project>

--- a/webui/portal/src/main/java/org/exoplatform/portal/application/PortalRequestContext.java
+++ b/webui/portal/src/main/java/org/exoplatform/portal/application/PortalRequestContext.java
@@ -42,6 +42,7 @@ import org.exoplatform.Constants;
 import org.exoplatform.commons.utils.ExpressionUtil;
 import org.exoplatform.commons.utils.PortalPrinter;
 import org.exoplatform.commons.xml.DOMSerializer;
+import org.exoplatform.services.security.ConversationState;
 import org.exoplatform.container.ExoContainer;
 import org.exoplatform.container.ExoContainerContext;
 import org.exoplatform.container.PortalContainer;
@@ -65,6 +66,7 @@ import org.exoplatform.services.log.Log;
 import org.exoplatform.services.resources.LocaleContextInfo;
 import org.exoplatform.services.resources.Orientation;
 import org.exoplatform.services.resources.ResourceBundleManager;
+import org.exoplatform.services.security.IdentityConstants;
 import org.exoplatform.web.ControllerContext;
 import org.exoplatform.web.application.JavascriptManager;
 import org.exoplatform.web.application.URLBuilder;
@@ -110,8 +112,6 @@ public class PortalRequestContext extends WebuiRequestContext {
     public static final String REQUEST_TITLE = "portal:requestTitle".intern();
 
     public static final String REQUEST_METADATA = "portal:requestMetadata".intern();
-
-    private static final String LAST_PORTAL_NAME = "prc.lastPortalName";
 
     private static final String DO_LOGIN_PATTERN = "login";
 
@@ -295,17 +295,18 @@ public class PortalRequestContext extends WebuiRequestContext {
     }
 
     public UserPortalConfig getUserPortalConfig() {
+        String remoteUser = null;
         if (userPortalConfig == null) {
-            String remoteUser = getRemoteUser();
-            SiteType siteType = getSiteType();
+            ConversationState conversationState = ConversationState.getCurrent();
+            if (conversationState != null
+              && conversationState.getIdentity() != null
+              && !IdentityConstants.ANONIM.equals(conversationState.getIdentity().getUserId())) {
+              remoteUser = conversationState.getIdentity().getUserId();
+            }
 
             String portalName = getCurrentPortalSite();
-            HttpSession session = request_.getSession();
             try {
                 userPortalConfig = portalConfigService.getUserPortalConfig(portalName, remoteUser, PortalRequestContext.USER_PORTAL_CONTEXT);
-                if (userPortalConfig != null) {
-                    session.setAttribute(LAST_PORTAL_NAME, portalName);
-                }
             } catch (Exception e) {
                 return null;
             }
@@ -316,12 +317,8 @@ public class PortalRequestContext extends WebuiRequestContext {
 
     private String getCurrentPortalSite() {
       String portalName = null;
-      HttpSession session = request_.getSession();
       if (SiteType.PORTAL == getSiteType()) {
         portalName = getSiteName();
-      }
-      if (portalName == null && session != null) {
-        portalName = (String) session.getAttribute(LAST_PORTAL_NAME);
       }
       if (portalName == null) {
         portalName = portalConfigService.getMetaPortal();

